node/rpc/src/lib.rs

```diff
@@ -88,492 +49,204 @@ where
-                                            let msg = e.message();
-                                            send_error(3, &ws_sender, msg).await;
-                                            return;
-                                        }
-
-                                        let ws_sender = ws_sender.clone();
-                                        task::spawn(async move {
-                                            while let Some(event) = head_changes.next().await {
-                                                let data = StreamingData {
-                                                    json_rpc: "2.0",
-                                                    method: "xrpc.ch.val",
-                                                    params: (
-                                                        chain_notify_count_curr,
-                                                        vec![HeadChangeJson::from(&event)],
-                                                    ),
-                                                };
-
-                                                if let Err(e) =
-                                                    send_response(&ws_sender, &data).await
-                                                {
-                                                    let msg = e.message();
-                                                    send_error(3, &ws_sender, msg).await;
-                                                }
-                                            }
+    let rpc_server = Arc::new(
+        Server::new()
+            .with_data(Data(state))
+            // Auth API
+            .with_method("Filecoin.AuthNew", auth_new::<DB, KS, B>)
+            .with_method("Filecoin.AuthVerify", auth_verify::<DB, KS, B>)
+            // Chain API
+            .with_method(
+                "Filecoin.ChainGetMessage",
+                chain_api::chain_get_message::<DB, KS, B>,
+            )
+            .with_method("Filecoin.ChainReadObj", chain_read_obj::<DB, KS, B>)
+            .with_method("Filecoin.ChainHasObj", chain_has_obj::<DB, KS, B>)
+            .with_method(
+                "Filecoin.ChainGetBlockMessages",
+                chain_block_messages::<DB, KS, B>,
+            )
+            .with_method(
+                "Filecoin.ChainGetTipsetByHeight",
+                chain_get_tipset_by_height::<DB, KS, B>,
+            )
+            .with_method("Filecoin.ChainGetGenesis", chain_get_genesis::<DB, KS, B>)
+            .with_method(
+                "Filecoin.ChainTipSetWeight",
+                chain_tipset_weight::<DB, KS, B>,
+            )
+            .with_method("Filecoin.ChainGetTipSet", chain_get_tipset::<DB, KS, B>)
+            .with_method(
+                "Filecoin.ChainHeadSubscription",
+                chain_head_sub::<DB, KS, B>,
+            )
+            .with_method("Filecoin.ChainNotify", chain_notify::<DB, KS, B>)
```

Comment:
You defined a const for this name, but you dont use it here. Same with the other new ones you've defined, like ChainHeadSubscription

Reply:
Yeah, it seemed inconsistent with the pattern used for the other methods on the router. I think maybe I should refactor all the method names out in a more deliberate and well-thought out way when working on the client library, which would be a separate PR.